climpred/relative_entropy.py

```diff
@@ -0,0 +1,301 @@
+
+
+def compute_relative_entropy(initialized, control,
+                             anomaly_data=False, neofs=None, curv=True,
+                             ntime=None, detrend_by_control_unitialized=True,
+                             nmember_control=10):
+    """
+    Compute relative entropy.
+
+    Calculates EOFs from anomalies. Projects fields on EOFs to receive
+    pseudo-Principle Components per initialization and lead year. Calculate
+    relative entropy based on _relative_entropy_formula.
+
+    Args:
+        initialized (xr.Dataset): anomaly ensemble data with dimensions
+                                    initialization, member, time and
+                                    spatial [lon (x), lat(y)].
+                                    DPLE or PM_ds
+        control (xr.Dataset): anomaly control distribution with
+                                              non-spatial dimensions:
+                                              spatial [lon (x), lat(y)].
+                                              - LENS: member, time
+                                              - PM_control: initialization
+        anomaly_data (bool): Input data is anomaly alread. Default: False.
+        neofs (int): number of EOFs to use.
+                     Default: initialized.member.size.
+        curv (bool): if curvilinear grids disables EOF weights.
+        ntime (int): number of timesteps calculated.
+        nmember_control (int): number of members created from
+                               bootstrapping from control
+
+    Returns:
+        rel_ent (xr.Dataset): relative entropy
+
+    """
+    # Defaults
+    if neofs is None:
+        neofs = initialized.member.size
+    if ntime is None:
+        ntime = initialized.time.size
+
+    # case if you submit control with dim time and member, LENS case
+    if 'member' in control.dims:
+        control_uninitialized = _bootstrap_dim(
+            control, initialized.time.size, time_dim='time',
+            dim='initialization',
+            dim_label=list(initialized.initialization.values))
+
+    # case if you only submit control with dim time, PM case
+    else:
+        control_uninitialized = xr.concat([
+            _bootstrap_dim(control,
+                           initialized.time.size, dim='member',
+                           dim_label=np.arange(nmember_control))
+            for _ in range(initialized.initialization.size)],
+            dim='initialization')
+        control_uninitialized['initialization'] = initialized.initialization.values
+
+    # initialized and control_uninitialized are allowed to have different
+    # dims as I need more members to sample my control distr. properly
+    if set(initialized.dims) == set(control_uninitialized.dims):
+        warnings.warn(
+            "Warning: initialized and control_uninitialized have different coords.")
+
+    # the datasets should only contain one variable
+    if isinstance(control_uninitialized, xr.Dataset):
```

Comment:
I think these will *always* be datasets, so you shouldn't need this if statement.